src/AESecb.cpp

```diff
@@ -0,0 +1,144 @@
+/**
+  @file AESecb.cpp
+  Implementation of ECB mode of operation for AES-128 algorithm
+*/
+#include "AESecb.hpp"
+#include <iostream>
+
+
+void printEncryptOutput(std::vector<unsigned char>& output) {
+  std::cout << std::endl;
+  std::cout << "Encrypt result";
+  for (int i = 0; i < output.size() / 16; i++){
+    std::cout << std::endl;
+    std::cout << "Block " << i << ": ";
+    for (int j = 0; j < 16; j++) {
+      std::cout << std::hex << (int) output[j+(i*16)];
+      std::cout << " ";
+    }
+  }
+}
+
+
+/**
+  Implements cipher with ECB mode of operation
+  @param input: array of hex values representing plaintext
+  @param output: array of hex values representing (padded) ciphertext
+  @param key: key to use
+  @param keysize: size of the key
+  @return array of decrypted values
+*/
+void encrypt_ecb(std::array<unsigned char, 16>& input, std::vector<unsigned char>& output, unsigned char* key, unsigned int keysize) {
+
+  // Calculate padding length, then copy input array and padding into plaintext
+  const int inputSize = input.size();
```

Comment:
https://wiki.sei.cmu.edu/confluence/display/cplusplus/CTR50-CPP.+Guarantee+that+container+indices+and+iterators+are+within+the+valid+range 
This recommends that indices into these containers be of type std::size_t to avoid the possibility of negative numbers.

Reply:
Updated this in AESecb.cpp. Haven't updated this in the encrypt and decrypt files, but that can be part of a different pull request for secure coding.